clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -20,9 +27,12 @@ import org.broadinstitute.clio.util.model.UpsertId
   */
 class ElasticsearchIndex[CI <: ClioIndex](
   val clioIndex: CI,
-  private[elasticsearch] val fieldMapper: ElasticsearchFieldMapper
+  private[elasticsearch] val defaultFields: Json = Json.obj(),
+  private[elasticsearch] val fieldMapper: ElasticsearchFieldMapper,
+  private[elasticsearch] val additionalIdFields: Seq[String] = Seq.empty
```

Comment:
I think we should derive this from `defaultFields`, instead of leaving it wide open. The only time we should be shimming in defaults is to modify the `_id` value (otherwise we can use normal upserts), so I'd rather it be locked down.

Reply:
Also it appears to be unused in this commit; I'm not sure what purpose it serves. What is the case where need a field that's part of the ID but is not part of the key?